examples/finnegans_wake_demo/finnegans-wake-demo-lynx.py

```diff
@@ -0,0 +1,191 @@
+
+ nucypher is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Affero General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License
+ along with nucypher.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+import datetime
+import maya
+import os
+from umbral.keys import UmbralPublicKey
+from web3.main import Web3
+
+from nucypher.blockchain.eth.signers.base import Signer
+from nucypher.characters.lawful import Alice, Bob, Ursula
+from nucypher.characters.lawful import Enrico as Enrico
+from nucypher.crypto.powers import SigningPower, DecryptingPower
+from nucypher.utilities.logging import GlobalLoggerSettings
+
+# Twisted Logger
+GlobalLoggerSettings.set_log_level(log_level_name='debug')
+GlobalLoggerSettings.start_console_logging()
+
+# Execute the download script (download_finnegans_wake.sh) to retrieve the book
+BOOK_PATH = os.path.join('.', 'finnegans-wake.txt')
+
+# Change this value to to perform more or less total re-encryptions
+# in order to avoid processing the entire book's text. (it's long)
+NUMBER_OF_LINES_TO_REENCRYPT = 25
+
+# Seednode
+SEEDNODE_URI = "https://lynx.nucypher.network:9151"
+SEEDNODE = Ursula.from_seed_and_stake_info(SEEDNODE_URI)
+
+######################
+# Boring setup stuff #
+######################
+
+# Configuration
+DOMAIN = 'lynx'  # testnet
+
+# Replace with ethereum RPC endpoint
+ETH_PROVIDER = ''
+PROVIDER_URI = os.environ.get('NUCYPHER_PROVIDER_URI', ETH_PROVIDER)
+
+# Replace with signer URI
+WALLET_FILEPATH = ''
+ETH_WALLET = f'keystore://{WALLET_FILEPATH}'
+SIGNER_URI = os.environ.get('NUCYPHER_SIGNER_URI', ETH_WALLET)
+
+# Replace with alice's ethereum address
+ETH_ADDRESS = ''
+ALICE_ETH_ADDRESS = os.environ.get('ALICE_ETH_ADDRESS', ETH_ADDRESS)
+
+
+#####################
+# Bob the BUIDLer  ##
+#####################
+
+# First there was Bob.
+BOB = Bob(
+    known_nodes=[SEEDNODE],
+    domain=DOMAIN,
+    provider_uri=PROVIDER_URI
+)
+
+# Bob gives his public keys to alice.
+verifying_key = BOB.public_keys(SigningPower)
+encrypting_key = BOB.public_keys(DecryptingPower)
+
+######################################
+# Alice, the Authority of the Policy #
+######################################
+
+# Alice ethereum wallet
+wallet = Signer.from_signer_uri(SIGNER_URI)
+password = input(f'Enter password to unlock {ALICE_ETH_ADDRESS}: ')
+wallet.unlock_account(account=ALICE_ETH_ADDRESS, password=password)
+
+ALICE = Alice(
+
+    # Connection details
+    domain=DOMAIN,
+    known_nodes=[SEEDNODE],
+    provider_uri=PROVIDER_URI,
+
+    # Wallet details
+    checksum_address=ALICE_ETH_ADDRESS,
+    signer=wallet,
+    client_password=password
+)
+
+# Here are the Policy details.
+policy_end_datetime = maya.now() + datetime.timedelta(days=5)
+m, n = 2, 3
+label = b"secret/files/and/stuff"
```

Comment:
Is there any concern of reuse of the label through multiple runs of the demo?

Reply:
No, It's a new Alice and Bob every run.